### PR TITLE
feat(protocol): Simulate and plot l2 basefee

### PR DESCRIPTION
**Pull request by adaki2004**
Simulation for L2 base fee.

It seems it works as expected, but script and simulation needs to be adjusted when we are changing these:
```
GWEI = 1e9
ETHEREUM_TARGET = 15 * 1e6
ETHEREUM_BASE_FEE = 10 * GWEI
TAIKO_TARGET = ETHEREUM_TARGET * 10
TAIKO_BASE_FEE = ETHEREUM_BASE_FEE // 10
ADJUSTMENT_QUOTIENT = 8
ADJUSTMENT_FACTOR = TAIKO_TARGET * 8
```

### CODE REVIEW

File: packages/protocol/simulation/simulation_data_parser.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                data = line.rstrip().split(':')
+                AVG_GAS_USED_PER_L1_BLOCK = float(data[1]) / 1000000
+                continue
+
+            if AVERAGE_GAS_PRICE_IN_L2 in line:
+                data = line.rstrip().split(':')
+                AVG_BASEFEE_PER_L2_BLOCK = float(data[1]) / 1000000000
+                continue
+
+def parse_args():
```
Comment (dantaik) [inline packages/protocol/simulation/simulation_data_parser.py:138]: can we have default parameter values at the top of the file so we can easily modify there?

Comment (adaki2004) [reply packages/protocol/simulation/simulation_data_parser.py:138]: This is just a parser and plotter. Please modify `TaikoL2.t.sol`

Comment (dantaik) [reply packages/protocol/simulation/simulation_data_parser.py:138]: I try to run `python3 simulation_data_parser.py ./exports/simulation_data_XXXXXXXXXX.txt` but end up with error :

```
Traceback (most recent call last):
  File "/Users/d/Projects/taiko/taiko-mono/packages/protocol/simulation/simulation_data_parser.py", line 150, in <module>
    main()
  File "/Users/d/Projects/taiko/taiko-mono/packages/protocol/simulation/simulation_data_parser.py", line 146, in main
    parse_and_plot(inputs.file)
  File "/Users/d/Projects/taiko/taiko-mono/packages/protocol/simulation/simulation_data_parser.py", line 62, in parse_and_plot
    with open(filename) as file:
```

File: packages/protocol/simulation/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Simulation and Exported Data Parser and Visualization
+
+## How to use
+
+### Prerequisites
+
+For running the `simulation_data_parser.py`, you need to install the required plugins: `(python3 -m pip install -r requirements.txt)`
```
Comment (dantaik) [inline packages/protocol/simulation/README.md:7]: I think we should git ingore all generated files.

File: packages/protocol/simulation/simulation_data_parser.py (hunk 1 above)
Comment (adaki2004) [reply packages/protocol/simulation/simulation_data_parser.py:138]: Replace `XXXX`.

File: packages/protocol/simulation/README.md (hunk 2 above)
Comment (adaki2004) [reply packages/protocol/simulation/README.md:7]: Feel free to modify this branch accordingly.

**Review by dantaik — APPROVED** [review]

**Review by dantaik — APPROVED** [review]

**Review by Brechtpd — APPROVED** [review]

**Pull request merged**